pulsar/schema.go

```diff
@@ -219,6 +226,107 @@ func (ps *ProtoSchema) GetSchemaInfo() *SchemaInfo {
 	return &ps.SchemaInfo
 }
 
+type ProtoNativeSchema struct {
+	SchemaInfo
+}
+
+func NewProtoNativeSchemaWithMessage(message proto.Message, properties map[string]string) *ProtoNativeSchema {
+	return NewProtoNativeSchema(GetProtoNativeSchemaInfo(message), properties)
+}
+
+func NewProtoNativeSchema(protoNativeSchemaDef string, properties map[string]string) *ProtoNativeSchema {
```

Comment:
I think this function should not be exposed to users. It's hard for users to know which string should be passed to the 1st parameter. I suggest only exposing the method that accepts a `proto.Message`:

```go
func NewProtoNativeSchema(message proto.Message, properties map[string]string) *ProtoNativeSchema {
        return newProtoNativeSchema(GetProtoNativeSchemaInfo(message), properties)
}

func newProtoNativeSchema(protoNativeSchemaDef string, properties map[string]string) *ProtoNativeSchema {
```

Reply:
The `GetProtoNativeSchemaInfo` and `GetFileProto` functions should not be exposed as well.

Reply:
Fixed, thanks.